File: project/server/Models/Locations/DTOs/LocationDto.cs
using System.Text.Json.Serialization;

namespace server.Models.Locations.DTOs
{
	[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
	[JsonDerivedType(typeof(SinglelineLocationDto), "Singleline")]
	[JsonDerivedType(typeof(MultilineLocationDto), "Multiline")]
	[JsonDerivedType(typeof(FileLocationDto), "File")]
	[JsonDerivedType(typeof(ProjectLocationDto), "Project")]
	public abstract class LocationDto
	{
		public Guid Id { get; set; }
		public string FilePath { get; set; } = string.Empty;
	}
}

File: client/src/types/dtos/LocationDto.ts
import type { CommentType } from "./CommentType";

export default interface LocationDto {
	id: string | null;
	type: CommentType;
	filePath: string;
	// Singleline
	lineNumber?: number;
	// Multiline
	startLineNumber?: number;
	endLineNumber?: number;
	// File and Project
	description?: string;
}
